components/ui/data-table/data-table-expanded-row.tsx

```diff
@@ -0,0 +1,51 @@
+import { flexRender } from "@tanstack/react-table";
+import {
+  TableCell,
+  TableRow,
+} from "@/components/ui/table";
+import { Row } from "@tanstack/react-table";
+
+interface DataTableExpandedRowProps<TData extends Record<string, unknown>> {
+  row: Row<Record<string, unknown>>;
+  data: TData[];
+  generateExpandedRowStatement: (input: {data: TData[], row: Row<Record<string, unknown>>}) => string;
+}
+
+export const DataTableExpandedRow = <TData extends Record<string, unknown>>(props: DataTableExpandedRowProps<TData>) => {
+  const remainingCell = props.row.getVisibleCells().at(0);
+
+  if(!remainingCell) {
+    return null;
+  }
+
+  return (
+    <TableRow
+      data-state={props.row.getIsSelected()}
+      onClick={() => props.row.getToggleExpandedHandler()()}
+    >
+      <TableCell colSpan={(props.row.getVisibleCells().length)} className="p-0">
+        <div className="
+          flex
+          w-auto
+          overflow-x-hidden
+          max-w-[95vw]
+          lg:max-w-[79vw]
+          sticky
+          left-0
+        "
+        >
+          <div
+            className="p-4 border-2"
+          >
+            {flexRender(remainingCell.column.columnDef.cell, remainingCell.getContext())}
+          </div>
```

Comment:
It's a div not a `td` so that's why the width doesn't match the width of the table header. I can't make this both a `<td>` and have it be sticky so it's a bit of a concession I had to make.

![image](https://github.com/serotracker/dashboard/assets/86806388/cfe89817-943b-412a-8a23-a1a84bb72095)